### PR TITLE
Add unit test suite for pure logic functions

### PR DESCRIPTION
**Pull request by roshkhatri**
#### Summary
Adds a test suite covering the pure logic functions across the codebase. The repo previously had zero tests. This gives us a safety net for catching regressions as we add new features.

What's covered
289 unit tests across 12 test modules, targeting functions that don't require a running Valkey server or external services:

- test_cpu_utils.py — parse_core_range, calculate_cpu_ranges, validate_explicit_cpu_ranges
- test_compare_benchmark.py — calculate_mean, calculate_stdev, calculate_confidence_interval, calculate_prediction_interval, calculate_percentage_change, average_multiple_runs, discover_config_keys, group_by_command
- test_deep_merge.py — deep_merge flat/nested merging and override precedence
- test_benchmark_config.py — validate_config, parse_bool, validation helpers, validate_cpu_allocation, validate_test_groups, _get_active_ports
- test_metrics_processor.py — MetricsProcessor.create_metrics with valid, empty, and malformed data
- test_benchmark_command.py — ClientRunner._build_benchmark_command across TLS, CPU pinning, duration modes
- test_csv_parsing.py — _parse_csv_row, _find_csv_start
- test_scenario_expansion.py — _expand_scenario_options
- test_postgres_utils.py — _is_list_subset, _is_config_subset, _is_config_array_subset, detect_field_type, analyze_metrics_schema, convert_metrics_to_rows
- test_server_utils.py — ServerLauncher._parse_cluster_info
- test_client_runner_logic.py — additional ClientRunner logic tests

#### Changes
- conftest.py
-  — shared fixtures (minimal_valid_config, minimal_test_groups_config, sample_benchmark_data, minimal_client_runner) and sys.path setup
- tests/test_*.py — 12 test modules
- requirements-test.txt — pytest>=7.0, hypothesis>=6.0

#### Running
```
pip install -r requirements-test.txt
pytest
```

### CODE REVIEW

File: tests/test_compare_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+        assert calculate_stdev([]) == 0.0
+
+    def test_normal_list(self):
+        result = calculate_stdev([2.0, 4.0, 4.0, 4.0, 5.0, 5.0, 7.0, 9.0])
+        assert result > 0.0
+
+    def test_identical_values(self):
+        assert calculate_stdev([3.0, 3.0, 3.0]) == 0.0
+
+    def test_filters_none(self):
```
Comment (rainsupreme) [inline tests/test_compare_benchmark.py]: Could add a test with None and multiple valid values to make sure it still calculates std dev among the valid values correctly

File: tests/test_deep_merge.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+from valkey_benchmark import deep_merge
+
+# ---------------------------------------------------------------------------
+# Flat dict merge with override precedence
+# ---------------------------------------------------------------------------
+
+
+class TestDeepMergeFlatDicts:
+    """Validates: Requirement 5.1"""
```
Comment (rainsupreme) [inline tests/test_deep_merge.py]: delete these orphaned references to test requirements

File: tests/test_metrics_processor.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+    def test_writes_to_new_directory(self, processor, tmp_path):
+        results_dir = tmp_path / "new_results"
+        new_metrics = [{"command": "GET", "rps": 100000}]
+
+        processor.write_metrics(results_dir, new_metrics)
+
+        metrics_file = results_dir / "metrics.json"
+        assert metrics_file.exists()
+
+        import json
```
Comment (rainsupreme) [inline tests/test_metrics_processor.py]: import json once at top of file

File: requirements-test.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+pytest>=7.0
+hypothesis>=6.0
```
Comment (VoletiRam) [inline requirements-test.txt]: Are we using hypothesis?

File: tests/test_server_utils.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    )
+
+
+# ---------------------------------------------------------------------------
+# _parse_cluster_info — valid inputs
+# ---------------------------------------------------------------------------
+
+
+class TestParseClusterInfoValid:
+    """Validates: Requirement 13.1"""
```
Comment (rainsupreme) [inline tests/test_server_utils.py]: there are requirements mentioned here too

File: .github/workflows/tests.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```
[the remark this answers is not in the document]
Comment (rainsupreme) [reply .github/workflows/tests.yml]: Also, this requirements caching thing could be used in the other workflows too, right? (If it's worth doing, worth doing every time?)

**Review by rainsupreme — APPROVED** [review]
Looking good! Just had one thing about dependency caching in the workflow - not critical but would be nice to fix.
